app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/MongoSendTask.java

```diff
@@ -16,19 +17,42 @@
  */
 public class MongoSendTask extends AsyncTask<String, Void, Void> {
         private Context context;
-        public List<BgSendQueue> bgsQueue = new ArrayList<BgSendQueue>();
+        //public List<BgSendQueue> bgsQueue = new ArrayList<BgSendQueue>();
         public List<CalibrationSendQueue> calibrationsQueue = new ArrayList<CalibrationSendQueue>();
 
+        PowerManager.WakeLock wakeLock;
+        private static int lockCounter = 0;
         private Exception exception;
         private static final String TAG = MongoSendTask.class.getSimpleName();
 
         public MongoSendTask(Context pContext) {
-            calibrationsQueue = CalibrationSendQueue.mongoQueue();
-            bgsQueue = BgSendQueue.mongoQueue();
             context = pContext;
+            PowerManager pm = (PowerManager) context.getSystemService(Context.POWER_SERVICE);
+            wakeLock = pm.newWakeLock(PowerManager.PARTIAL_WAKE_LOCK, "WifiReader");
+            wakeLock.acquire();
+            lockCounter++;
+            Log.e(TAG,"MongosendTask - wakelock acquired " + lockCounter);
+
         }
 
         public Void doInBackground(String... urls) {
+            boolean sendMore = true;
+            try {
+                while (sendMore) {
```

Comment:
If there is data waiting there is nothing wrong with sending it, but the uploader should probably register for network state/connectivity broadcasts and perform big batch backfills only while on unmetered connections, preferably while plugged in.  It should also implement exponential backoff to avoid excessive retries, especially if the reattempts are very frequent.  I intentionally left it gated on new readings being available because the mobile radio will be woken up already, so extra data isn't very expensive power-wise.